dotnet/SimpleSauce.Test/SauceOptionsTests.cs

```diff
@@ -1,13 +1,386 @@
+using System;
+using System.Collections.Generic;
 using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
 using OpenQA.Selenium.Edge;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.IE;
+using OpenQA.Selenium.Safari;
 using Simple.Sauce;
 
 namespace SimpleSauce.Test
 {
     [TestClass]
     public class SauceOptionsTests : BaseTest
     {
+        [TestInitialize]
+        public void Setup()
+        {
+            SauceOptions = new SauceOptions();
+        }
+
+        [TestMethod]
+        public void UsesLatestChromeWindowsVersionsByDefault()
+        {
+            SauceOptions.BrowserName.Should().BeEquivalentTo("chrome");
+            SauceOptions.BrowserVersion.Should().BeEquivalentTo("latest");
+            SauceOptions.PlatformName.Should().BeEquivalentTo("Windows 10");
+        }
+
+        [TestMethod]
+        public void UpdatesBrowserBrowserVersionPlatformVersion()
+        {
+            SauceOptions.BrowserVersion = "68";
+            SauceOptions.BrowserName = Browsers.Firefox;
+            SauceOptions.PlatformName = Platforms.Mac.HighSierra;
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo("firefox");
+            SauceOptions.BrowserVersion.Should().BeEquivalentTo("68");
+            SauceOptions.PlatformName.Should().BeEquivalentTo("macOS 10.13");
+        }
+
+        /*
+         * Safari is an intersting and frustrating animal because you can only use
+         * a specific safariVersion with a specific platformVersion
+         * In a previous iteration, I created something that I really like.
+         * A good way for the user to avoid having to know which PlatformName needs to
+         * go with which BrowserVersion. I really like that protection and that API.
+         * However, I'm cool with simplifying the API and allowing 
+         *   SauceOptions.BrowserVersion = "68";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4779740)